crates/compiler/builtins/roc/Decode.roc

```diff
@@ -96,3 +97,6 @@ fromBytes = \bytes, fmt ->
                     Err TooShort -> Err TooShort
             else
                 Err (Leftover rest)
+
+mapResult : DecodeResult a, (a -> b) -> DecodeResult b
+mapResult = \{ result, rest }, mapper -> { result: Result.map result mapper, rest }
```

Comment:
🤔 If we had tuples, would this just return a tuple? In which case, would people just use something like `Tuple.map`?

Reply:
Personally I think I'd prefer a record with named fields here anyway - otherwise I could see how it might be kind of confusing that the second element of the tuple would refer to the unparsed bytes, and not e.g. the whole buffer or the parsed bytes. I could see arguments either way though.

Something like `Tuple.map : (a, b), (a -> c), (b -> d) -> (c, d)` and/or `Tuple.mapFirst (a, )fields, (a -> b) -> (b, )fields` seems super useful though (if we have open tuples)

Reply:
I was thinking about that being potentially confusing, but:
* I think if I'm doing parsing stuff, I'll get used to that very quickly and then remember it
* The two tuple fields will have different types (after all, why would anyone parse a `List U8` into `List U8`?) so if you mess it up, the compiler will tell you